includes/class-woocommerce.php

```diff
@@ -660,6 +660,16 @@ public function wpdb_table_fix() {
 			$wpdb->tables[]             = 'woocommerce_termmeta';
 		}
 	}
+	
+	/**
+	 * Generates user agent
+	 *
+	 * @param string    $webhook Optional webhook (eg Hookshot).
+	 * @return string
+	 */
+	public static function user_agent_header( $webhook = '' ) {
+		return sprintf( 'WooCommerce/%s %s (compatible; ClassicCommerce/%s)', WC()->version, $webhook, WC()->cc_version );
```

Comment:
When `$webhook` is left as the default `''` the user-agent header will have two consecutive spaces in it. Maybe something like this instead?

```
if ( ! empty( $webhook ) ) {
    $webhook = ' ' . $webhook;
}
return sprintf( 'WooCommerce/%s%s (compatible; ClassicCommerce/%s)', WC()->version, $webhook, WC()->cc_version );
```

Other than that minor issue (unlikely to cause any problems I think), this looks good to me :+1:

Reply:
That is true. I'll correct. Again 😄